src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtFactory.java

```diff
@@ -116,11 +195,7 @@ private static SignedJWT createSignedES256Jwt(
 
       var algorithm = JWSAlgorithm.EdDSA;
       var type = JOSEObjectType.JWT;
-
-      JWSHeader.Builder jwsHeaderBuilder =
-          new JWSHeader.Builder(algorithm).type(type).keyID(issuer).base64URLEncodePayload(true);
-
-      var header = jwsHeaderBuilder.build();
+      var header = new JWSHeader.Builder(algorithm).type(type).keyID(issuer + "#" + keyId).build();
```

Comment:
Why is the `base64URLEncodePayload` removed?

Reply:
I havn't test it yet but i suppose from the name of it, it will encode the payload into base64, and we want it to stay as json when encode the jwt

Reply:
This has to do with the payload being "detached". If we don't want it to be "detached", it must then be base64urlencoded beforehand. This won't be reflected in any tests I think, but please add it back and double-check.

Reply:
okay, checked from the docs and added it back